soh/soh/Enhancements/item-tables/ItemTableManager.cpp

```diff
@@ -0,0 +1,59 @@
+#include "ItemTableManager.h"
+
+ItemTableManager::ItemTableManager() {
+}
+
+ItemTableManager::~ItemTableManager() {
+    this->itemTables.clear();
+}
+
+bool ItemTableManager::AddItemTable(uint16_t tableID) {
+    ItemTable newItemTable;
+    return itemTables.emplace(tableID, newItemTable).second;
+}
+
+bool ItemTableManager::AddItemEntry(uint16_t tableID, uint16_t getItemID, uint16_t itemID, uint16_t objectID, int16_t drawID, uint16_t textID, uint16_t field, bool chestAnim) {
+    ItemTable* itemTable = RetrieveItemTable(tableID);
+    if (itemTable == nullptr) {
+        return false;
+    }
+    GetItemEntry getItemEntry = GET_ITEM(itemID, objectID, drawID, textID, field, chestAnim);
+    return itemTable->emplace(getItemID, getItemEntry).second;
+}
+
+bool ItemTableManager::AddItemEntry(uint16_t tableID, uint16_t getItemID, GetItemEntry getItemEntry) {
+    ItemTable* itemTable = RetrieveItemTable(tableID);
+    if (itemTable == nullptr) {
+        return false;
+    }
+    return itemTable->emplace(getItemID, getItemEntry).second;
+}
```

Comment:
Just wanted to have a way to do it that didn't require GET_ITEM in case there were situations where it was difficult to use for some reason. But I'm not using said function anywhere so I could remove it.